Assets/Scripts/Die.cs

```diff
@@ -0,0 +1,224 @@
+using UnityEngine;
+using UnityEngine.Rendering;
+using System;
+using UnityEngine.Rendering.HighDefinition;
+using System.Collections;
+
+public class Die : Singleton<Die>
```

Comment:
I do not like the class name Die.  i would call it somthing like:
DeathHandler
DeathManager
EndGameLogic
DeathAction
DeathEvent